GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -110,8 +110,11 @@ public void removeSceneObject(GVRSceneObject sceneObject) {
      * Remove all scene objects.
      */
     public void removeAllSceneObjects() {
+        getMainCameraRig().removeAllChildren();
         mSceneObjects.clear();
         NativeScene.removeAllSceneObjects(getNative());
+
+        addSceneObject(getMainCameraRig().getOwnerObject());
```

Comment:
i'm not fully following this.  probably because it is late.  i guess you remove all sceneobjects from the camera rig, except for the camera sceneobjects, then call clear(), then add back the camera rig?  and somehow clear() doesn't remove the camera sceneobjects that were added to the list of sceneobjects in the scene?